### DISCUSSION TITLE
middleware with jwt token from backend services

### DISCUSSION THREAD

**Discussion by ehsanrezaeee** [amannn/next-intl#821] [General]
when i am trying to use jwt token that i receive from my backend golang and set them in cookies, i use Next js middleware to protect my routes. however there is a missing docs about using cookie authorization and redirecting user to specific route using NextResponse.redirect() with next intl and its middleware

invoking NextResponse.redirect(new URL()) wont redirect users. how can i use NextResponse with next intl createmiddleware() to redirect user to specific path?

**Comment by gsdev01** [comment]
Hi, any update on this ?

**Reply by ehsanrezaeee** [reply]
hope this help:

```typescript
 if (
   some authentication happened!
  ) {
    if (request.nextUrl.pathname !== "/auth") {
      return NextResponse.redirect(new URL("/auth", request.url));
    }
  }

  const handleI18nRouting = createMiddleware({
    locales,
    localeDetection: false,
    localePrefix,
    defaultLocale: "fa",
  });

  let response = handleI18nRouting(request);

  return response;
}
```
